client/session.go

```diff
@@ -128,12 +120,12 @@ func NewSession(
 		ProviderKeyPin:      s.provider.IdentityKey,
 		LinkKey:             s.linkKey,
 		LogBackend:          logBackend,
-		PKIClient:           pkiCacheClient,
+		PKIClient:           pkiClient,
 		OnConnFn:            s.onConnection,
 		OnMessageFn:         s.onMessage,
 		OnACKFn:             s.onACK,
 		OnDocumentFn:        s.onDocument,
-		DialContextFn:       proxyCfg.ToDialContext("authority"),
+		DialContextFn:       proxyCfg.ToDialContext(string(idHash[:])),
```

Comment:
This will get passed in plaintext to Tor/socks, maybe we shouldn't leak our internal identifiers to the network.
Since this is only initialized once per session, it seems safer to just generate a random string?
Then we can close https://github.com/katzenpost/katzenpost/issues/56

Reply:
good catch 153f4b6b37892eaa7036b3329a47798977c80347